Update v1.1 executors playbook (#3950)
sourcegraph/handbook: content/departments/engineering/dev/admin-exp/devops/managed/v1.1/mi1-1_enable_executors_process.md

### Confirm executors is actually working

> Then run the check, which will increase the minimal replica of executors auto scaling group to `1` and retry until either the executors are up or a 5 minute timeout has been reached, after which you should check configuration errors:

Reviewer: This is a little confusing since it says "run the check" but doesn't specify what that means. Presumably it means run `mg executors check` but the way the sentence is written
> after which you should check configuration errors

makes it seem like `mg executors check` should only be run afterwards.

Reviewer: I think more clearly describing exactly what to run here would make sense.

e.g. "Run `mg executors check` which will increase the count ... After it's complete, rerun `mg executors check` to check for configuration errors"

> Then run the check, which will increase the minimal replica of executors auto scaling group to `1` and retry until either the executors are up or a 5 minute timeout has been reached, after which you should check configuration errors:

Author: ```suggestion
Then run the command below, which will 
- increase the minimal replica of executors auto scaling group to `1`
- retry until either the executors are up or a 5 minute timeout has been reached, after which you should check configuration errors if it fails
- scale auto scaling group back to `0`
```

@danieldides how about this?

Reviewer: That's much more clear to me, thanks!

![create-budget-04](https://storage.googleapis.com/sourcegraph-assets/create-executor-budgets-04.png)

![create-budget-02](https://storage.googleapis.com/sourcegraph-assets/create-executor-budgets-02.png)

## Troubleshooting

Ensure instance groups are there

```sh
$ gcloud compute instance-groups list --zones=$zone --project=$PROJECT
NAME                             LOCATION       SCOPE  NETWORK                MANAGED  INSTANCES
batches--sourcegraph-executor    us-central1-a  zone   sourcegraph-executors  Yes      0
codeintel--sourcegraph-executor  us-central1-a  zone   sourcegraph-executors  Yes      0
```

Ensure `minNumReplicas` is greater than `0`

```sh
$ gcloud compute instance-groups managed describe batches--sourcegraph-executor --zone=$ZONE --project=$PROJECT --format=json | jq '.autoscaler.autoscalingPolicy'
{
  "minNumReplicas": 1,
}
```

```sh
$ gcloud compute instance-groups managed describe codeintel--sourcegraph-executor --zone=$ZONE --project=$PROJECT --format=json | jq '.autoscaler.autoscalingPolicy'
{
  "minNumReplicas": 1,
}
```

> Ensure there is an active instance belong to one of the instance group (notes `batches--sourcegraph-executor-rqfs`)

Author: ```suggestion
Ensure there is an active instance belong to one of the instance group (notes the `batches--sourcegraph-executor-rqfs` instance). Sometimes it may take GCP longer to spawn a new instance, so be patient. If no new instance is created for an unreasonable amount of time, consult GCP documentation for next step.
```

![create-budget-04](https://storage.googleapis.com/sourcegraph-assets/create-executor-budgets-04.png)

![create-budget-02](https://storage.googleapis.com/sourcegraph-assets/create-executor-budgets-02.png)

## Troubleshooting

Ensure instance groups are there

```sh
$ gcloud compute instance-groups list --zones=$zone --project=$PROJECT
NAME                             LOCATION       SCOPE  NETWORK                MANAGED  INSTANCES
batches--sourcegraph-executor    us-central1-a  zone   sourcegraph-executors  Yes      0
codeintel--sourcegraph-executor  us-central1-a  zone   sourcegraph-executors  Yes      0
```

Ensure `minNumReplicas` is greater than `0`

```sh
$ gcloud compute instance-groups managed describe batches--sourcegraph-executor --zone=$ZONE --project=$PROJECT --format=json | jq '.autoscaler.autoscalingPolicy'
{

>   "minNumReplicas": 0,

Author: ```suggestion
  "minNumReplicas": 1,
```

}
```

```sh
$ gcloud compute instance-groups managed describe codeintel--sourcegraph-executor --zone=$ZONE --project=$PROJECT --format=json | jq '.autoscaler.autoscalingPolicy'
{

>   "minNumReplicas": 0,

Author: ```suggestion
  "minNumReplicas": 1,
```
